tests/Unit/TestApplicationTrait.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Composer package "cpsit/auditor".
+ *
+ * Copyright (C) 2021 Elias Häußler <e.haeussler@familie-redlich.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+namespace CPSIT\Auditor\Tests\Unit;
+
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Trait for usage of test application in Unit tests.
+ *
+ * @author Elias Häußler <e.haeussler@familie-redlich.de>
+ * @license GPL-2.0-or-later
+ */
+trait TestApplicationTrait
```

Comment:
I'm not quite sure what's the purpose of this trait. As far as I can see. those are functional tests not unit tests. When unit testing file system operation a mock (like mikey179/vfsstream) should be used.

Reply:
My idea was to not mock the Composer environment but rather use a real testing application for it, because I think that the usage of a real environment makes it easier to capture errors produced by any Composer operation rather than a mocked Composer environment.

For this, I originally tried to use `mikey179/vfsstream`, but it turned out that the Composer factory cannot handle file streams `vfs://foo/baz` but only real file systems. That's why I decided to add a tiny testing application and use ist as template for relevant test cases. The result was this `TestApplicationTrait`.

Do you have suggestions on this? Maybe it's not worth the effort and we should rather stick to mocks of the Composer environment here? I am afraid that using mocked Composer environments won't identify all possible problems created by this plugin. Since the successful usage of Composer operations for this plugin are very essential, I'd try to run tests as close as possible to real Composer environments. Maybe it's better to write functional tests then? What do you think?